bundles/persistence/org.openhab.persistence.h2sql/java/org/openhab/persistence/h2sql/internal/H2SqlPersistenceService.java

```diff
@@ -0,0 +1,464 @@
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.Dictionary;
+import java.util.Formatter;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.openhab.core.items.GroupItem;
+import org.openhab.core.items.Item;
+import org.openhab.core.items.ItemNotFoundException;
+import org.openhab.core.items.ItemRegistry;
+import org.openhab.core.library.items.ColorItem;
+import org.openhab.core.library.items.ContactItem;
+import org.openhab.core.library.items.DateTimeItem;
+import org.openhab.core.library.items.DimmerItem;
+import org.openhab.core.library.items.NumberItem;
+import org.openhab.core.library.items.RollershutterItem;
+import org.openhab.core.library.items.SwitchItem;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.HSBType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.OpenClosedType;
+import org.openhab.core.library.types.PercentType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.persistence.FilterCriteria;
+import org.openhab.core.persistence.FilterCriteria.Ordering;
+import org.openhab.core.persistence.HistoricItem;
+import org.openhab.core.persistence.PersistenceService;
+//import org.openhab.core.persistence.PersistentStateRestorer;
+import org.openhab.core.persistence.QueryablePersistenceService;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.osgi.service.cm.ConfigurationException;
+import org.osgi.service.cm.ManagedService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+/**
+ * This is the implementation of the H2 SQL {@link PersistenceService}.
+ * See http://h2database.com
+ * 
+ * H2 database licensed under EPL (http://h2database.com/html/license.html)
+ * 
+ * Data is persisted with the following conversions -:
+ * 
+ * Item-Type         Data-Type    H2SQL-Type
+ * =========         =========    ==========
+ * ColorItem         HSBType      CHAR(25)
+ * ContactItem       OnOffType    CHAR(6)
+ * DateTimeItem      DateTimeType DATETIME
+ * DimmerItem        PercentType  TINYINT
+ * NumberItem        DecimalType  DOUBLE
+ * RollershutterItem PercentType  TINYINT
+ * StringItem        StringType   VARCHAR(20000)
+ * SwitchItem        OnOffType    CHAR(3)
+ * 
+ * In the store method, type conversion is performed where the default type for
+ * an item is not as above For example, DimmerType can return OnOffType, so to
+ * keep the best resolution, we store as a number in SQL and convert to
+ * DecimalType before persisting to H2SQL.
+ * 
+ * @author Chris Jackson - Initial contribution
+ * @since 1.8.0
+ */
+public class H2SqlPersistenceService implements QueryablePersistenceService, ManagedService {
```

Comment:
since update() is empty there is no more need to implement ManagedService, please also remove update() and the reference in OSGI-INF